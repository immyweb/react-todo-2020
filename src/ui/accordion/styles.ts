import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
