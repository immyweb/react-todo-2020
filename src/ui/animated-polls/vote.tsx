import * as React from 'react';

import { OptionBar } from './option-bar';
import {
  APVote,
  APVoteImgBkgnd,
  APVoteHeadline,
  APCopy,
  APVoteSubHeadline,
  APIntroLink,
  APIntroHashTag,
  APVoteContent,
  APVoteImgBkgndCover,
  APVoteImgBkgndCoverGradient,
} from './styles';

import { IVoteScreen, ICard } from './types';

interface IProps extends IVoteScreen {
  options: ICard[];
  clickVote: (cardIds: string[]) => void;
}

export const VoteScreen: React.FC<IProps> = ({
  bkgndImgMobile,
  bkgndImgDesktop,
  title,
  introTxt,
  ctaTxt,
  cta,
  ctaLink,
  ctaUrl,
  options,
  theme,
  clickVote,
}) => {
  return (
    <APVote>
      <APVoteImgBkgnd imageMobile={bkgndImgMobile} imageDesktop={bkgndImgDesktop} />
      <APVoteImgBkgndCover />
      <APVoteImgBkgndCoverGradient />
      <APVoteContent>
        <APVoteHeadline>{title}</APVoteHeadline>
        <APCopy textAlign={'left'}>{introTxt}</APCopy>
        {options.map((option, i) => (
          <OptionBar key={i} clickVote={clickVote} {...option} {...theme} />
        ))}
        <APVoteSubHeadline>{ctaTxt}</APVoteSubHeadline>
        <APCopy>
          <APIntroLink href={ctaUrl}>
            {cta} <APIntroHashTag>{ctaLink}</APIntroHashTag>
          </APIntroLink>
        </APCopy>
      </APVoteContent>
    </APVote>
  );
};
