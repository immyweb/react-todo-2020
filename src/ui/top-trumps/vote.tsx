import * as React from 'react';

import { IVoteScreen } from './types';
import { ITheme } from '../../types/theme';
import {
  TTDefaultInner,
  TTVoteButtonContainer,
  TTVoteButton,
  TTVoteButtonThumb,
  TTVoteButtonThumbContainer,
  TTVoteButtonText,
  TTWrapper,
} from './styles';

interface IProps extends IVoteScreen {
  className: string;
  theme: ITheme;
  clickVote: (cardIds: string[]) => void;
}

export const Vote: React.SFC<IProps> = ({ theme, cards, clickVote, imageMobile, imageDesktop }) => {
  return (
    <TTDefaultInner imageMobile={imageMobile} imageDesktop={imageDesktop}>
      <TTWrapper>
        <TTVoteButtonContainer>
          {cards.map(({ id, fName, sName, imageThumb }) => (
            <TTVoteButton
              key={sName}
              onClick={() => clickVote([id])}
              colour={theme.primary}
              hoverBackgroundColour={theme.secondary}
            >
              <TTVoteButtonThumbContainer>
                <TTVoteButtonThumb src={imageThumb} />
              </TTVoteButtonThumbContainer>
              <TTVoteButtonText colour={theme.secondary}>{`${fName} ${sName}`}</TTVoteButtonText>
            </TTVoteButton>
          ))}
        </TTVoteButtonContainer>
      </TTWrapper>
    </TTDefaultInner>
  );
};
