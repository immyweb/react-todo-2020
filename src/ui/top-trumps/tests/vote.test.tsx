import * as React from 'react';
import { shallow } from 'enzyme';

import { Vote } from '../vote';
import { TTVoteButton } from '../styles';

const theme = {
  primary: '#ffffff',
  secondary: '#000000',
};

const cards = [
  {
    id: 'wilder',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg',
    fName: 'Deontay',
    sName: 'Wilder',
  },
  {
    id: 'whyte',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg',
    fName: 'Dillian',
    sName: 'Whyte',
  },
  {
    id: 'fury',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg',
    fName: 'Tyson',
    sName: 'Fury',
  },
];

function shallowRender(props?: any) {
  return shallow(
    <Vote
      className={'youCantBuy'}
      theme={theme}
      cards={cards}
      headline={'Its time to decide'}
      subHeadline={'Would should Jamie fight next?'}
      bottomCopy={'Cast your vote now'}
      imageMobile={'https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg'}
      imageDesktop={'https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg'}
      clickVote={() => {}}
      {...props}
    />,
  );
}

describe('<Vote />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct number of vote buttons', () => {
    const wrapper = shallowRender();
    const items = wrapper.find(TTVoteButton);
    expect(items.length).toEqual(cards.length);
  });

  it('calls clickVote with correct id when user votes', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ clickVote: fn });
    const btn = wrapper.find(TTVoteButton).first();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
    expect(fn.mock.calls[0][0][0]).toBe('wilder');
    const btn2 = wrapper.find(TTVoteButton).last();
    btn2.simulate('click');
    expect(fn).toHaveBeenCalledTimes(2);
    expect(fn.mock.calls[1][0][0]).toBe('fury');
  });
});
