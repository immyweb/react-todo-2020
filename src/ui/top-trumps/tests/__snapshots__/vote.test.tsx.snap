// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Vote /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    bottomCopy="Cast your vote now"
    cards={
      Array [
        Object {
          "fName": "Deontay",
          "id": "wilder",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg",
          "sName": "Wilder",
        },
        Object {
          "fName": "Dillian",
          "id": "whyte",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg",
          "sName": "Whyte",
        },
        Object {
          "fName": "Tyson",
          "id": "fury",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg",
          "sName": "Fury",
        },
      ]
    }
    className="youCantBuy"
    clickVote={[Function]}
    headline="Its time to decide"
    imageDesktop="https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg"
    imageMobile="https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg"
    subHeadline="Would should Jamie fight next?"
    theme={
      Object {
        "primary": "#ffffff",
        "secondary": "#000000",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <glamorous(section)>
        <glamorous(div)>
          <glamorous(a)
            colour="#ffffff"
            hoverBackgroundColour="#000000"
            onClick={[Function]}
          >
            <glamorous(div)>
              <glamorous(img)
                src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
              />
            </glamorous(div)>
            <glamorous(div)
              colour="#000000"
            >
              Deontay Wilder
            </glamorous(div)>
          </glamorous(a)>
          <glamorous(a)
            colour="#ffffff"
            hoverBackgroundColour="#000000"
            onClick={[Function]}
          >
            <glamorous(div)>
              <glamorous(img)
                src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
              />
            </glamorous(div)>
            <glamorous(div)
              colour="#000000"
            >
              Dillian Whyte
            </glamorous(div)>
          </glamorous(a)>
          <glamorous(a)
            colour="#ffffff"
            hoverBackgroundColour="#000000"
            onClick={[Function]}
          >
            <glamorous(div)>
              <glamorous(img)
                src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
              />
            </glamorous(div)>
            <glamorous(div)
              colour="#000000"
            >
              Tyson Fury
            </glamorous(div)>
          </glamorous(a)>
        </glamorous(div)>
      </glamorous(section)>,
      "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg",
      "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <glamorous(div)>
          <glamorous(a)
            colour="#ffffff"
            hoverBackgroundColour="#000000"
            onClick={[Function]}
          >
            <glamorous(div)>
              <glamorous(img)
                src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
              />
            </glamorous(div)>
            <glamorous(div)
              colour="#000000"
            >
              Deontay Wilder
            </glamorous(div)>
          </glamorous(a)>
          <glamorous(a)
            colour="#ffffff"
            hoverBackgroundColour="#000000"
            onClick={[Function]}
          >
            <glamorous(div)>
              <glamorous(img)
                src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
              />
            </glamorous(div)>
            <glamorous(div)
              colour="#000000"
            >
              Dillian Whyte
            </glamorous(div)>
          </glamorous(a)>
          <glamorous(a)
            colour="#ffffff"
            hoverBackgroundColour="#000000"
            onClick={[Function]}
          >
            <glamorous(div)>
              <glamorous(img)
                src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
              />
            </glamorous(div)>
            <glamorous(div)
              colour="#000000"
            >
              Tyson Fury
            </glamorous(div)>
          </glamorous(a)>
        </glamorous(div)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Deontay Wilder
              </glamorous(div)>
            </glamorous(a)>,
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Dillian Whyte
              </glamorous(div)>
            </glamorous(a)>,
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Tyson Fury
              </glamorous(div)>
            </glamorous(a)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Wilder",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <glamorous(div)>
                  <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
                  />
                </glamorous(div)>,
                <glamorous(div)
                  colour="#000000"
                >
                  Deontay Wilder
                </glamorous(div)>,
              ],
              "colour": "#ffffff",
              "hoverBackgroundColour": "#000000",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "src": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Deontay Wilder",
                  "colour": "#000000",
                },
                "ref": null,
                "rendered": "Deontay Wilder",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Whyte",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <glamorous(div)>
                  <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
                  />
                </glamorous(div)>,
                <glamorous(div)
                  colour="#000000"
                >
                  Dillian Whyte
                </glamorous(div)>,
              ],
              "colour": "#ffffff",
              "hoverBackgroundColour": "#000000",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "src": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Dillian Whyte",
                  "colour": "#000000",
                },
                "ref": null,
                "rendered": "Dillian Whyte",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Fury",
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <glamorous(div)>
                  <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
                  />
                </glamorous(div)>,
                <glamorous(div)
                  colour="#000000"
                >
                  Tyson Fury
                </glamorous(div)>,
              ],
              "colour": "#ffffff",
              "hoverBackgroundColour": "#000000",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "src": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Tyson Fury",
                  "colour": "#000000",
                },
                "ref": null,
                "rendered": "Tyson Fury",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <glamorous(section)>
          <glamorous(div)>
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Deontay Wilder
              </glamorous(div)>
            </glamorous(a)>
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Dillian Whyte
              </glamorous(div)>
            </glamorous(a)>
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Tyson Fury
              </glamorous(div)>
            </glamorous(a)>
          </glamorous(div)>
        </glamorous(section)>,
        "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg",
        "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <glamorous(div)>
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Deontay Wilder
              </glamorous(div)>
            </glamorous(a)>
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Dillian Whyte
              </glamorous(div)>
            </glamorous(a)>
            <glamorous(a)
              colour="#ffffff"
              hoverBackgroundColour="#000000"
              onClick={[Function]}
            >
              <glamorous(div)>
                <glamorous(img)
                  src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
                />
              </glamorous(div)>
              <glamorous(div)
                colour="#000000"
              >
                Tyson Fury
              </glamorous(div)>
            </glamorous(a)>
          </glamorous(div)>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(a)
                colour="#ffffff"
                hoverBackgroundColour="#000000"
                onClick={[Function]}
              >
                <glamorous(div)>
                  <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
                  />
                </glamorous(div)>
                <glamorous(div)
                  colour="#000000"
                >
                  Deontay Wilder
                </glamorous(div)>
              </glamorous(a)>,
              <glamorous(a)
                colour="#ffffff"
                hoverBackgroundColour="#000000"
                onClick={[Function]}
              >
                <glamorous(div)>
                  <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
                  />
                </glamorous(div)>
                <glamorous(div)
                  colour="#000000"
                >
                  Dillian Whyte
                </glamorous(div)>
              </glamorous(a)>,
              <glamorous(a)
                colour="#ffffff"
                hoverBackgroundColour="#000000"
                onClick={[Function]}
              >
                <glamorous(div)>
                  <glamorous(img)
                    src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
                  />
                </glamorous(div)>
                <glamorous(div)
                  colour="#000000"
                >
                  Tyson Fury
                </glamorous(div)>
              </glamorous(a)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Wilder",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <glamorous(div)>
                    <glamorous(img)
                      src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
                    />
                  </glamorous(div)>,
                  <glamorous(div)
                    colour="#000000"
                  >
                    Deontay Wilder
                  </glamorous(div)>,
                ],
                "colour": "#ffffff",
                "hoverBackgroundColour": "#000000",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <glamorous(img)
                      src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "src": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Deontay Wilder",
                    "colour": "#000000",
                  },
                  "ref": null,
                  "rendered": "Deontay Wilder",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Whyte",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <glamorous(div)>
                    <glamorous(img)
                      src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
                    />
                  </glamorous(div)>,
                  <glamorous(div)
                    colour="#000000"
                  >
                    Dillian Whyte
                  </glamorous(div)>,
                ],
                "colour": "#ffffff",
                "hoverBackgroundColour": "#000000",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <glamorous(img)
                      src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "src": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Dillian Whyte",
                    "colour": "#000000",
                  },
                  "ref": null,
                  "rendered": "Dillian Whyte",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Fury",
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <glamorous(div)>
                    <glamorous(img)
                      src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
                    />
                  </glamorous(div)>,
                  <glamorous(div)
                    colour="#000000"
                  >
                    Tyson Fury
                  </glamorous(div)>,
                ],
                "colour": "#ffffff",
                "hoverBackgroundColour": "#000000",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": <glamorous(img)
                      src="https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "src": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Tyson Fury",
                    "colour": "#000000",
                  },
                  "ref": null,
                  "rendered": "Tyson Fury",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
